src/FirmwarePlugin/PX4/PX4FirmwarePlugin.cc

```diff
@@ -789,3 +742,35 @@ const QVariantList& PX4FirmwarePlugin::toolIndicators(const Vehicle* vehicle)
 
     return _toolIndicatorList;
 }
+
+void PX4FirmwarePlugin::updateAvailableFlightModes(FlightModeList modeList)
+{
+    _availableFlightModeList.clear();
+    for(auto mode: modeList){
+
+        // Update Multi Rotor
+        switch (mode.custom_mode) {
```

Comment:
Both of these switches would be better without a default. By listing out every mode if/when a new mode gets added the compiler will complain if these switches are not updated. With the default, a new mode could be added and if updating these switches are forgotten they will fall to default and possible do the wrong thing.

Reply:
So in the default case should we give any pop up message to UI?

Reply:
There shouldn't be a default case at all. The compiler won't allow builds with missing cases.